docs/vm-configuration.md

```diff
@@ -266,3 +266,227 @@ the generation of multiple libvirt XML documents, the domain XML and associated
 documents for secrets, network filters and node devices. So there is not a strict
 1-1 mapping between the k8s API resource representation and the libvirt low
 level representation.
+
+Mapping rules for mapping libvirt XML to the KubeVirt API
+---------------------------------------------------------
+
+In order to support libvirt features, we tend to expose the libvirt API as much
+as possible in a 1:1 fashion. However, there are some differences which need to
+be taken into account:
+
+ 1. Our API is json/yaml and not xml.
+ 2. Host specifics are not exposed on the API.
+ 3. We want to make use of the golang xml/yaml/json mappers as much as possible.
+ 4. Ordering of devices can be important (e.g. for pci address assignment).
```

Comment:
can be important, but it should not: it's more fragile if ordering influences addressing. Which use case do you have in mind?

Reply:
@michalskrivanek that is your area.

Reply:
just for future reference - with libvirt enhancement for naming devices we now have a way how to uniquely identify the right device (#528)